Make compile with Qt5 and use Qt5 by default.

gassmoeller opened the pull request:
This PR fixes #15, and also makes Qt5 the default version to use (it has be out for a long time, even Ubuntu 14.04 optionally packages 5.2, although 4.8 is still the default). If Qt5 is not found it falls back to Qt4.

@ian-r-rose: If you have a few minutes, could you give this PR a try? I tested it with Qt 5.2 (packaged with Ubuntu 14.04), and Qt 5.9 (current version, self-compiled), but I would like to have one more data point.

gassmoeller committed:
Make compile with and use Qt5 by default.

ian-r-rose commented:
I can confirm that this compiles with Qt 5.6.2 on Ubuntu 16.10. Thanks @gassmoeller!

bangerth commented on CMakeLists.txt:
```diff
@@ -68,3 +88,7 @@ EXPORT(TARGETS parameter_gui_exe
   )
+
+IF(${Qt5_FOUND})
+  qt5_use_modules (parameter_gui_exe Core Gui Xml Widgets)
```
Can you use upper casee letters for the command here as usual?

bangerth commented on browse_lineedit.cpp:
```diff
@@ -13,11 +13,13 @@
-
 #include "browse_lineedit.h"
 
+#if QT_VERSION >= 0x050000
```
Where is that preprocessor define set?

gassmoeller commented:
Thanks @ian-r-rose for testing.
@bangerth: I addressed your comment. QT_VERSION is set inside of QtGlobal, which is a header that is included by all Qt header files. Therefore it is defined as soon as any other Qt header is included. Should I make that explicit by including QtGlobal before using QT_VERSION?

bangerth commented:
Yes, I think that would make it obvious. Ok to merge with that.

gassmoeller committed:
Address comments. Silence warning.

gassmoeller commented:
Addressed your comments. Also silenced a warning for cmake version >3.0 if Qt5 can not be found. Also since including <QtWidgets> or <QtGui> essentially includes all existing widgets, while we need only a few, I replaced that include by individual includes for the used Widgets. The only place with the general include left is the main window (which would need a long list of includes otherwise). 
Ready to merge from my perspective (but I have no merge permissions).

bangerth commented:
Thanks for cleaning up the code while you're at it!
